File: src/main/java/com/celestia/interfaces/IGLProgramBuilder.java

package com.celestia.interfaces;

import com.celestia.models.GLProgram;
import com.jogamp.opengl.GLAutoDrawable;

public interface IGLProgramBuilder {
    public IGLProgramBuilder addShader(String shaderPath, int shaderType);
    public GLProgram build(GLAutoDrawable glAutoDrawable);
}

File: scala/src/main/scala/com/celestia/interfaces/IGLProgramBuilder.scala

package com.celestia.interfaces

import com.jogamp.opengl.{GL3, GL2ES2, GL4, GLAutoDrawable}
import com.celestia.models.GLProgram

/**
  * Interface for creating a GL Program Builder class. 
  * This interface defines how the program will interact
  * with GL Programs; by adding a collection of shaders
  * to a program and compiling them on the GPU. 
  */
trait IGLProgramBuilder {
  def addShader(shaderPath:String, shaderType:Int):IGLProgramBuilder
  def build(gl:GLAutoDrawable):GLProgram
}
